src/NetMQ/NetMQPoller.cs

```diff
@@ -344,10 +401,13 @@ public void Remove([NotNull] Socket socket)
                 throw new ArgumentNullException(nameof(socket));
             CheckDisposed();
 
-            Run(() =>
+            RunAsync(() =>
```

Comment:
Is there a reason you didn't create `RemoveAsync(Socket)`?

Reply:
I did, line [307.](https://github.com/zeromq/netmq/blob/455f6b066c6a775f91e51e4a2c88879bbe2b86c5/src/NetMQ/NetMQPoller.cs#L307).

The crrent implementation was intentionally done to keep current behavior when calling the public API's original methods.  The only change from master I made here was `Run()`->`RunAsync()` in an effort to minimize my changes. 

We *could* call `RemoveAsync(socket)` here, to remove a secondary call chain? That's your call as the owner.  I think it is probably a good idea...

Reply:
Deleted first response, I misread.  You are talking about `System.Net.Sockets.Socket` type here.  I didn't even realize it was available...  So that's why I didn't add the method.

Never occurred to me to use a raw socket with this lib...

Reply:
Added overloaded `RemoveAsync()`